addOns/openapi/src/main/java/org/zaproxy/zap/extension/openapi/ImportDialog.java

```diff
@@ -0,0 +1,402 @@
+                                        "ZAP-OpenAPI-UI-Import")
+                                .start();
+                    });
+        }
+        return buttonImport;
+    }
+
+    private JProgressBar getProgressBar() {
+        if (progressBar == null) {
+            progressBar = new JProgressBar();
+            progressBar.setIndeterminate(true);
+            progressBar.setVisible(false);
+        }
+        return progressBar;
+    }
+
+    private void showProgressBar(boolean show) {
+        if (getProgressBar().isVisible() == show) {
+            return;
+        }
+
+        setSize(getWidth(), getHeight() + (show ? 10 : -10));
+        getProgressBar().setVisible(show);
+
+        getImportButton().setEnabled(!show);
+        getDefinitionField().setEnabled(!show);
+        getTargetField().setEnabled(!show);
+        getChooseFileButton().setEnabled(!show);
+        getContextsComboBox().setEnabled(!show);
+    }
+
+    private boolean importDefinition() {
+        String definitionLocation = getDefinitionField().getText();
+        if (definitionLocation == null || definitionLocation.isEmpty()) {
+            ThreadUtils.invokeAndWaitHandled(
+                    () -> {
+                        showWarningDialog(
+                                Constant.messages.getString(
+                                        MESSAGE_PREFIX + "error.missingDefinition"));
+                        getDefinitionField().requestFocusInWindow();
+                    });
+            return false;
+        }
+
+        try {
+            new URL(definitionLocation).toURI();
+            var uri = new URI(definitionLocation, true);
+            return extOpenApi.importOpenApiDefinition(
+                            uri, getTargetField().getText(), true, getSelectedContextId())
+                    != null;
+        } catch (URIException | MalformedURLException | URISyntaxException ignored) {
+            // Not a valid URI, try to import as a file
+        } catch (InvalidUrlException e) {
+            ThreadUtils.invokeAndWaitHandled(
+                    () -> {
+                        showWarningInvalidUrl(e.getUrl());
+                        getDefinitionField().requestFocusInWindow();
+                    });
+            return false;
+        } catch (OpenApiExceptions.EmptyDefinitionException e) {
+            ThreadUtils.invokeAndWaitHandled(
+                    () -> {
+                        showWarningDialog(e.getLocalizedMessage());
+                        getDefinitionField().requestFocusInWindow();
+                    });
+            return false;
+        }
+
+        var file = new File(definitionLocation);
+        if (!file.canRead()) {
+            ThreadUtils.invokeAndWaitHandled(
+                    () -> {
+                        showWarningDialog(
+                                Constant.messages.getString(
+                                        MESSAGE_PREFIX + "error.fileNotFound", definitionLocation));
```

Comment:
"openapi.import.error.fileNotFound", could be extracted a method to avoid duplication.